sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -44,31 +43,17 @@ public class UnusedPrivateMember : SonarDiagnosticAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
 
-        private static readonly Accessibility maxAccessibility = Accessibility.Private;
-
-        private ConcurrentBag<Diagnostic> possibleUnusedInternalMembers;
-
         protected sealed override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterCompilationStartAction(
                 c =>
                 {
-                    var shouldRaise = true;
-                    possibleUnusedInternalMembers = new ConcurrentBag<Diagnostic>();
-
-                    c.RegisterSemanticModelAction(
-                        cc =>
-                        {
-                            var isInternalsVisibleToAttributeFound = cc.SemanticModel.SyntaxTree.GetRoot()
-                                .DescendantNodes()
-                                .OfType<AttributeListSyntax>()
-                                .SelectMany(list => list.Attributes)
-                                .Any(a => IsInternalVisibleToAttribute(a, cc.SemanticModel));
-                            if (isInternalsVisibleToAttributeFound)
-                            {
-                                shouldRaise = false;
-                            }
-                        });
+                    // Collect potentially removable internal members from the project to evaluate when
+                    // the compilation is over, depending on whether InternalsVisibleTo attribute is present
+                    // or not.
+                    var removableInternalMembers = new ConcurrentBag<ISymbol>();

```

Comment:
If where only interested in removable internal _types_ then this variable should be renamed.